lib/csv_helper/loader.rb

```diff
@@ -73,5 +54,22 @@ def merge_options(options)
       options.reverse_merge(key_mapping: key_mapping, value_converters: value_converters)
              .reverse_merge(SMARTER_CSV_OPTIONS)
     end
+
+    # default validations are run during import to prevent bad data from going into table,
+    # this runs validations with a context of :load_csv
+    # validations with this context are ran now due either the validation needing data in the table being loaded into
+    # or the validation causes issues on save
+    def load_csv_validation(records, failed_instances, options)
+      # Since row indexes start at 0 and spreadsheets on line 1,
+      # add 1 for the difference in indexes and 1 for the header row itself.
+      row_offset = CSV_FIRST_LINE + (options[:skip_lines] || 0)
+
+      records.each_with_index do |record, index|
+        unless record.valid?(:load_csv)
+          record.errors.add(:row, "Line #{index + row_offset}")
+          failed_instances << record
```

Comment:
You can change this to
```suggestion
          failed_instances << record if record.persisted?
```
This will prevent duplicates of the records that were marked invalid during the import from being added to the array.  Then you can remove the `.uniq` call in `uploads_controller#create`

Reply:
changed to `unless record.valid?(:after_import) && !record.persisted?` instead as to prevent an additional error being added to the record if not persisted